### PR TITLE
Update `lyon` to `0.17` in `iced_graphics`

### PR DESCRIPTION
**Pull request by hecrj**
Updates `lyon` to the latest release (`0.17`).

I have not reviewed all of the API changes yet, so there may be opportunities for further improvements.

### CODE REVIEW

File: graphics/src/widget/canvas/frame.rs [hunk 1]
```diff
@@ -127,12 +130,17 @@ impl Frame {
...
+        builder.add_rectangle(
             &lyon::math::Rect::new(top_left, size.into()),
-            &FillOptions::default().with_fill_rule(rule.into()),
-            &mut buffers,
-        )
-        .expect("Fill rectangle");
+            lyon::path::Winding::Positive,
+        );
+
+        let _ = builder.build().expect("Fill rectangle");
```
Comment (hecrj) [inline graphics/src/widget/canvas/frame.rs:143]: I'm noticing a performance regression of around an order of magnitude in the `game_of_life` example.

I imagine the new `Frame::fill_rectangle` implementation is the culprit. We should try to address this before merging.

**Comment by yamadapc** [comment]
One thing to try is `fill_options.with_intersections(false)` (though it's not new to 0.17).

I got around 2x improvement in `fill` with that path, I suppose there're places where it'd be safe to assume there're no intersections or to allow users to provide a hint to canvas that it's the case.

File: graphics/src/widget/canvas/frame.rs [hunk 2]
```diff
@@ -127,12 +130,17 @@ impl Frame {
                 lyon::math::Vector::new(size.width, size.height),
             );
 
-        let _ = lyon::tessellation::basic_shapes::fill_rectangle(
+        let mut tessellator = FillTessellator::new();
```
Comment (yamadapc) [inline graphics/src/widget/canvas/frame.rs:133]: This would also be retained in #988

File: graphics/src/widget/canvas/frame.rs (hunk 1 above)
Comment (hecrj) [reply graphics/src/widget/canvas/frame.rs]: The author of `lyon` addressed the issue in https://github.com/nical/lyon/pull/696 🎉

**Pull request merged**